app/controllers/tag_adjustments_controller.rb

```diff
@@ -3,7 +3,7 @@ def create
     authorize(User, :moderation_routes?)
     TagAdjustmentCreationService.new(
```

Comment:
There's a lingering issue here, it is probably present, or maybe I'm doing something wrong but I just triggered it twice and I want to ask your opinion.

This block calls `TagAdjustmentCreationService.new(params).create`, inside `TagAdjustmentCreationService` the `def create` calls `@tag_adjustment = TagAdjustment.create!(creation_args)` which means that if there's an error, it will propagate one level up and result in a 500 error without any information to the user. This is what happens to me in the logs:

```text
ActiveRecord::RecordInvalid (Validation failed: Tag can't be blank, Tag must exist)

ActiveRecord::RecordInvalid - Validation failed: Tag can't be blank, Tag must exist:
  app/services/tag_adjustment_creation_service.rb:8:in `create'
  app/controllers/tag_adjustments_controller.rb:11:in `create'
```

I triggered it by trying to add an already adjusted tag, or trying to remove an already removed tag.

Reply:
> I just added required: true to the tag_name and reason_for_adjustment to make sure they are filled out.. This will allow empty strings to get through but there are validations on the model.
> I could prevent this on the front-end with javascript but there are other object errors that are not being prevented or displayed. Right now if a mod enters a tag to remove that they are not moderating or that doesn't exist, it will cause a Validation failed: error. I could refactor to catch and display these errors to the new tag adjustment form on the mod view.
> 
> Let me know if you want me to refactor or leave as is?

This is what I was referring to in the comment up above.

@rhymes Thanks again for reviewing. I just pushed up a commit to catch and display these errors. I used redirect_to and flash message but I could user render to display the old form if preferred.  Let me know if you want me to make any changes.
![Screen Shot 2019-11-14 at 11 09 39 AM](https://user-images.githubusercontent.com/25860670/68897174-ccb45500-06e1-11ea-9c7b-f91c34fee4e5.png)

Reply:
There's a lingering issue here, it is probably present, or maybe I'm doing something wrong but I just triggered it twice and I want to ask your opinion.

This block calls TagAdjustmentCreationService.new(params).create, inside TagAdjustmentCreationService the def create calls @tag_adjustment = TagAdjustment.create!(creation_args) which means that if there's an error, it will propagate one level up and result in a 500 error without any information to the user. This is what happens to me in the logs:

ActiveRecord::RecordInvalid (Validation failed: Tag can't be blank, Tag must exist)

ActiveRecord::RecordInvalid - Validation failed: Tag can't be blank, Tag must exist:
  app/services/tag_adjustment_creation_service.rb:8:in `create'
  app/controllers/tag_adjustments_controller.rb:11:in `create'
I triggered it by trying to add an already adjusted tag, or trying to remove an already removed tag.

@rhymes I refactored this issue by saving the service object in a variable service = TagAdjustmentCreationService.new(so I could can call methods on it including service.tag_adjustment which adds calls TagAdjustment new instead of create.
@tag_adjustment ||= TagAdjustment.new(creation_args). I then check to see if it saves and if not, display the errors.
Let me know if you want me to change anything. Thanks